### DISCUSSION TITLE
how can I change the width of a Card？

### DISCUSSION THREAD

**Discussion by poppintiger** [zauberzeug/nicegui#1358] [Q&A]
### Description

When I change the w-32 to w-40, it turn out to be a card with very small width. Is there a good way to change the width?

```py
with ui.card().classes('w-32 h-32'):
    with ui.scroll_area():
        ui.chat_message('Hello NiceGUI!', name='Robot', stamp='now')
```

**Comment by falkoschindler** [comment]
You have plenty of options for setting the width of a UI element:
- You can use Tailwind classes like `w-60` or `w-80`. See https://tailwindcss.com/docs/width for a complete list.
- You can use custom values like `w-[200px]` or `w-[50vw]`.
- You can set the CSS property using the `style` method, e.g. `ui.card().style('width: 200px')`.

**Reply by steeeef99** [reply]
Hi could it be, that there is a maximum width for ui.cards?
I am trying to increase my cards width. But if I try to apply a width larger than 'w-96' or for example 'w-full' it is not applied / updated.

Thanks Stefan

**Reply by falkoschindler** [reply]
No, there is no upper limit. "w-full" should work as expected:
```py
ui.card().classes('w-full')
```

**Reply by rodja** [reply]
Tailwind only defines width up to `w-96` (see https://tailwindcss.com/docs/width). But you can put arbitrary width in `w-[XXXpx]`.

**Reply by steeeef99** [reply]
Hi thanks for the quick reply! In my case I use a ui.card in combination with an ui.dialog...
Could that be the Problem?

Here my Code:
```py
        with ui.dialog() as dialog, ui.card().classes('w-full'):
            ui.table(columns=self.columns_eqpinfo, rows=rows)
            ui.button('Close', on_click=dialog.close)

        dialog.open()
```

I also tried:
```py
        with ui.dialog().classes('w-full') as dialog, ui.card().classes('w-full'):
            ui.table(columns=self.columns_eqpinfo, rows=rows)
            ui.button('Close', on_click=dialog.close)

        dialog.open()
```

The width-settings are not applied, only if I use the w-96 (or smaller) settings from tailwind.
But the problem is My ui.card has to be wider than w-96...

**Reply by rodja** [reply]
You may be seeking the `maximized` prop: https://quasar.dev/vue-components/dialog#positioning, as we use it in the lightbox example:

https://github.com/zauberzeug/nicegui/blob/e926909f594bbd192ea54a4c5924e90ce01d0870/examples/lightbox/main.py#L15

**Reply by steeeef99** [reply]
Hi thanks for the replay, after I was investigating the created website my colleague noticed that there is a property in quasar.prod.css, that limits the max width.

After deactivating that option the ui.card is displayed as wanted :)

Unfortunately I dont know where to permanent disable that setting in the code.
I only managed to disable it in the browser Examine Tool...

`  @media (min-width: 600px)
.q-dialog__inner--minimized > div {
  max-width: 560px;
}`

**Reply by rodja** [reply]
You should use the props from the Dialog. Quasar does quite a lot of css magic.

**Reply by steeeef99** [reply]
Hi, thanks for the quick reply! I forgot to mention that your solution (using maximized) worked well :)
The problem that I was describing with the max width is only, when I want the ui.dialog has a fixed width.
For example 600px. That is not applied because of the problem I mentioned before, but I dont know how to fix yet

Greetings Stefan

**Reply by rodja** [reply]
Ah! Very interesting. The Quasar Dialog applies `max-width: 100%!important` to the inner child if prop `maximized` or `full-width` is set. Not elegant but working:

```py
with ui.dialog().props('full-width') as dialog, ui.row(), ui.card().classes('w-[600px] mx-auto'):
    ui.button('Close', on_click=dialog.close)
```

**Reply by steeeef99** [reply]
Hi,
thank you for your answer!
In my case it didnt worked as expected, because the content of the dialogue (in my case the table) overlaps the dialogue-window, thats is opend.

**Reply by steeeef99** [reply]
Hi rodja,
I checked your solution again and now it worked fine!
Sorry for the confusion!

Thank you very much!

best regards stefan
